### PR TITLE
search frontend: highlight query filter separator ':' specially

### PR DESCRIPTION
**Pull request by rvantonder**
It's prettier.

### Now

Screen Shot 2022-01-07 at 12 36 36 PM
Screen Shot 2022-01-07 at 12 36 19 PM

---

### Previously

Screen Shot 2022-01-07 at 12 55 50 PM
Screen Shot 2022-01-07 at 12 56 10 PM

### CODE REVIEW

File: client/shared/src/search/query/decoratedToken.ts [hunk 1]
```diff
@@ -1009,9 +1017,14 @@ export const decorate = (token: Token): DecoratedToken[] => {
...
                 type: 'field',
-                range: token.field.range,
+                range: { start: token.field.range.start, end: token.field.range.end - 1 },
                 value: token.field.value,
             })
+            decorated.push({
+                type: 'metaFilterSeparator',
+                range: { start: token.field.range.end, end: token.field.range.end + 1 },
+                value: ':',
+            })
```
Comment (rvantonder) [inline client/shared/src/search/query/decoratedToken.ts:1027]: Actual change for query bar highlighting

File: client/web/src/components/SyntaxHighlightedSearchQuery.tsx [hunk 2]
```diff
@@ -12,8 +12,9 @@ export const SyntaxHighlightedSearchQuery: React.FunctionComponent<{ query: stri
                       return (
                           <Fragment key={token.range.start}>
                               <span className="search-filter-keyword">
-                                  {query.slice(token.field.range.start, token.field.range.end)}:
+                                  {query.slice(token.field.range.start, token.field.range.end)}
                               </span>
+                              <span className="search-filter-separator">:</span>
```
Comment (rvantonder) [inline client/web/src/components/SyntaxHighlightedSearchQuery.tsx:17]: Actual change for query panel highlighting

File: client/web/src/search/input/SearchContextDropdown.tsx [hunk 3]
```diff
@@ -120,7 +120,8 @@ export const SearchContextDropdown: React.FunctionComponent<SearchContextDropdow
                 data-tooltip={disabledTooltipText}
             >
                 <code className={classNames('test-selected-search-context-spec', styles.buttonContent)}>
-                    <span className="search-filter-keyword">context:</span>
+                    <span className="search-filter-keyword">context</span>
+                    <span className="search-filter-separator">:</span>
```
Comment (rvantonder) [inline client/web/src/search/input/SearchContextDropdown.tsx:124]: Actual change for context dropdown menu

**Review by rrhyne — APPROVED** [review]
Very nice. :D

**Review by limitedmage — APPROVED** [review]

**Pull request merged**